rust/flatbuffers/src/builder.rs

```diff
@@ -235,6 +241,57 @@ impl<'fbb> FlatBufferBuilder<'fbb> {
         WIPOffset::new(o.value())
     }
 
+    #[inline]
+    pub fn create_shared_string<'a: 'b, 'b>(&'a mut self, s: &'b str) -> WIPOffset<&'fbb str> {
+        self.assert_not_nested(
+            "create_shared_string can not be called when a table or vector is under construction",
+        );
+        // Checks if strings_map is None to allocate it
+        if self.strings_map == None {
+            self.strings_map = Some(Vec::new())
+        }
+
+        // Saves a ref to owned_buf since rust doesnt like us refrencing it
+        // in the binary_search_by code.
+        let buf = &self.owned_buf;
+
+        let found = self.strings_map.as_ref().map(|pool| {
+
+            pool.binary_search_by(|offset| {
+                let ptr = offset.value() as usize;
+                // Gets The pointer to the size of the string
+                let str_memory = &buf[buf.len() - ptr..];
+                // Gets the size of the written string from buffer
+                let size = u32::from_le_bytes([
+                    str_memory[0],
+                    str_memory[1],
+                    str_memory[2],
+                    str_memory[3],
+                ]) as usize;
+                // Size of the string size
+                let string_size: usize = 4;
+                // Fetches actual string bytes from index of string after string size
+                // to the size of string plus string size
+                let iter = str_memory[string_size..size + string_size].iter();
+                // Compares bytes of fetched string and current writable string
+                iter.cloned().cmp(s.bytes())
+            })
+        });
+
```

Comment:
Okay, this should do the job! I had to make sure that everything is working and since we arent using the same logic as flexbuffers, I had to modify the code to make it work with flatbuffers. The idea of the code is pretty simple, we get the position of the string by getting the UOffset and subtracting the length of the current buffer. Then we take the size and read from the buffer and compare it.